crates/indexer/src/models/stake_models/staking_pool_voter.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+
+use super::stake_utils::StakeResource;
+use crate::{schema::current_staking_pool_voter, util::standardize_address};
+use aptos_api_types::{Transaction as APITransaction, WriteSetChange as APIWriteSetChange};
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+type StakingPoolAddress = String;
+pub type StakingPoolVoterMap = HashMap<StakingPoolAddress, CurrentStakingPoolVoter>;
+
+#[derive(Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(staking_pool_address))]
+#[diesel(table_name = current_staking_pool_voter)]
+pub struct CurrentStakingPoolVoter {
+    pub staking_pool_address: String,
+    pub voter_address: String,
+    pub last_transaction_version: i64,
+}
+
+impl CurrentStakingPoolVoter {
+    pub fn from_transaction(transaction: &APITransaction) -> anyhow::Result<StakingPoolVoterMap> {
+        let mut staking_pool_voters = HashMap::new();
+        let empty_change = vec![];
+        let (txn_version, changes) = match transaction {
+            APITransaction::UserTransaction(txn) => (txn.info.version.0 as i64, &txn.info.changes),
+            APITransaction::GenesisTransaction(txn) => {
+                (txn.info.version.0 as i64, &txn.info.changes)
+            }
+            APITransaction::BlockMetadataTransaction(txn) => {
```

Comment:
Voter cannot be changed in BlockMetadataTransaction

Reply:
There's no in case lolll. It's impossible to change a specific stake pool in blockmetadata tx as it doesn't have access to any signers.